lib/spack/spack/package_base.py

```diff
@@ -926,6 +922,32 @@ def global_license_file(self):
             self.global_license_dir, self.name, os.path.basename(self.license_files[0])
         )
 
+    # NOTE: return type should be Optional[Literal['all', 'specific', 'none']] in
+    # Python 3.8+, but we still support 3.6.
+    @property
+    def keep_werror(self) -> Optional[str]:
+        """Keep ``-Werror`` flags, matches ``config:flags:keep_werror`` to override config.
+
+        Valid return values are:
+        * ``"all"``: keep all ``-Werror`` flags.
+        * ``"specific"``: keep only ``-Werror=specific-warning`` flags.
+        * ``"none"``: filter out all ``-Werror*`` flags (this is the default).
+        * ``None``: respect the user's configuration.
```

Comment:
"none" is the configuration default, but not the default return value.  Maybe:

```suggestion
        * ``"none"``: filter out all ``-Werror*`` flags.
        * ``None``: respect the user's configuration (``"none"`` by default).
```